source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/BaseActionElementRenderer.java

```diff
@@ -266,47 +207,136 @@ private Activity getActivity(Context context)
             return null;
         }
 
-        @Override
-        public void onClick(View v) {
-            if (m_isInlineShowCardAction)
+        private void populateViewsDictionary()
+        {
+            m_viewDictionary = new HashMap<>();
+
+            ToggleVisibilityTargetVector toggleVisibilityTargetVector = m_toggleVisibilityAction.GetTargetElements();
+            View rootView = m_renderedAdaptiveCard.getView();
+
+            for (int i = 0; i < toggleVisibilityTargetVector.size(); ++i)
             {
-                Activity hostingActivity = getActivity(v.getContext());
-                if (hostingActivity != null) {
-                    View currentFocusedView = hostingActivity.getCurrentFocus();
-                    if (currentFocusedView != null) {
-                        currentFocusedView.clearFocus();
-                    }
+                ToggleVisibilityTarget target = toggleVisibilityTargetVector.get(i);
+                String elementId = target.GetElementId();
+
+                View foundView = rootView.findViewWithTag(new TagContent(elementId));
+                if (foundView != null)
+                {
+                    m_viewDictionary.put(elementId, foundView);
                 }
+            }
+        }
 
-                v.setPressed(m_invisibleCard.getVisibility() != View.VISIBLE);
-                for (int i = 0; i < m_hiddenCardsLayout.getChildCount(); ++i) {
-                    View child = m_hiddenCardsLayout.getChildAt(i);
-                    if (child != m_invisibleCard) {
-                        child.setVisibility(View.GONE);
-                    }
+        private void handleInlineShowCardAction(View v)
+        {
+            Activity hostingActivity = getActivity(v.getContext());
+            if (hostingActivity != null)
+            {
+                View currentFocusedView = hostingActivity.getCurrentFocus();
+                if (currentFocusedView != null)
+                {
+                    currentFocusedView.clearFocus();
+                }
+            }
+
+            v.setPressed(m_invisibleCard.getVisibility() != View.VISIBLE);
+            for (int i = 0; i < m_hiddenCardsLayout.getChildCount(); ++i)
```

Comment:
>++i [](start = 69, length = 3)

Nit: Consider using the more concise foreach equivalent syntax (i think for (View child : m_hiddenCardsLayout) ) ;

Reply:
Android API is weird in this kind of stuff so we have to get the elements one by one as it doesn't provide a "getChildren" method

---
In reply to: [277107266](https://github.com/Microsoft/AdaptiveCards/pull/2717#discussion_r277107266) [](ancestors = 277107266)

Reply:
Hmm you mean there is a requirement for being able to use the foreach java syntax for this collection? 

---
In reply to: [278224177](https://github.com/Microsoft/AdaptiveCards/pull/2717#discussion_r278224177) [](ancestors = 278224177,277107266)

Reply:
As far as I know, to use the foreach syntax you must have the collection or be able to get the collection with a getter method that returns the complete collection, in this case, Android only allows us to grab one element at a time by querying each index

[https://stackoverflow.com/questions/18668897/android-get-all-children-elements-of-a-viewgroup](https://stackoverflow.com/questions/18668897/android-get-all-children-elements-of-a-viewgroup) 

---
In reply to: [278771455](https://github.com/Microsoft/AdaptiveCards/pull/2717#discussion_r278771455) [](ancestors = 278771455,278224177,277107266)